### PR TITLE
feat(Modal): open programmatically

### LINKED ISSUE #66: [Modal] Open programmatically
### Description

Port of https://github.com/nuxt/ui/blob/dev/src/runtime/composables/useModal.ts

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by noook**
### 🔗 Linked issue

Resolves #66 

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality)
- [x] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Allows opening modals dynamically

~One feature is missing: we cannot detect the end of the animation to reset the modal state, I think we need something upstream on radix-vue~ 

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by benjamincanac** [comment]
We're no longer using a component to handle the transition but a CSS animation, I'm afraid it's not really related to `radix-vue` 🤔

**Comment by noook** [comment]
They use `Presence` which could kind of allow us to hook into the animation end (if there is one).

Submitted a PR (https://github.com/radix-vue/radix-vue/pull/901) on radix-vue so we can hook on it and reset state properly to avoid flickering while closing

**Comment by noook** [comment]
@benjamincanac It seems that the transition is now flawless, I didn't even update radix

I don't know what caused the issue last time I tried, but I remember the leaving animation was instant because the component was unmounted too early, it should be ready now

**Comment by benjamincanac** [comment]
Awesome! I've just renamed the `modals` plugin to `modal` and `ModalsProvider` component to `ModalProvider` for consistency.

File: src/runtime/providers/ModalProvider.vue [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<script lang="ts" setup>
```
Comment (benjamincanac) [inline src/runtime/providers/ModalProvider.vue]: Don't you think this should be in the `components/` dir? There might be some case where a user might need to use this.

Comment (noook) [reply src/runtime/providers/ModalProvider.vue]: bd62cf10

File: src/runtime/components/App.vue [hunk 2]
```diff
@@ -12,6 +12,7 @@ export interface ProviderProps extends ConfigProviderProps {
 import { toRef } from 'vue'
 import { ConfigProvider, TooltipProvider, useForwardProps } from 'radix-vue'
 import { reactivePick } from '@vueuse/core'
+import ModalProvider from '../providers/ModalProvider.vue'
 import { useId } from '#imports'
 import { UToaster } from '#components'
```
Comment (benjamincanac) [inline src/runtime/components/App.vue]:
```suggestion
import { useId } from '#imports'
import { UToaster, UModalProvider } from '#components'
```

File: src/runtime/components/App.vue [hunk 3]
```diff
@@ -31,5 +32,7 @@ const toasterProps = toRef(() => props.toaster)
         <slot />
       </UToaster>
     </TooltipProvider>
+
+    <ModalProvider />
```
Comment (benjamincanac) [inline src/runtime/components/App.vue]:
```suggestion
    <UModalProvider />
```

**Review by benjamincanac — APPROVED** [review]

**Comment by benjamincanac** [comment]
Thanks!!! 😊

**Pull request merged**